docs/machine-learning/how-to-guides/machine-learning-model-predictions-ml-net.md

```diff
@@ -84,7 +84,14 @@ HousingPrediction prediction = predictionEngine.Predict(inputData);
 
 If you access the `Score` property of the `prediction` object, you should get a value similar to `150079`.
 
-## Multiple predictions
+> [!TIP]
+> [`PredictionEngine`](xref:Microsoft.ML.PredictionEngine%602) is not thread-safe. Additionally, you have to create an instance of it everywhere it is needed within your application. As your application grows, this process can become unmanageable. For improved performance and thread safety, use a combination of dependency injection and the [PredictionEnginePool](xref:Microsoft.Extensions.ML.PredictionEnginePool%602) service, which creates an [`ObjectPool`](xref:Microsoft.Extensions.ObjectPool.ObjectPool%601) of [`PredictionEngine`](xref:Microsoft.ML.PredictionEngine%602) objects for use throughout your application.
```

Comment:
Is there something else in .NET that we could compare this to (like we usually compare MLContext to DBContext in EF) to make it more familiar for .NET devs?

Reply:
None that I can think of :/